File: apps/repo/dto/menu.go
package dto

type GetMenuReq struct {
	ID   int64  `json:"id"`
	Path string `json:"path"`
}

type MenuListReq struct {
	PageReq
	Path string `json:"path"`
	Name string `json:"name"`
}

type MenuItem struct {
	ID        int64          `json:"id"`        // ID
	ParentID  int64          `json:"parent_id"` // 父ID
	Path      string         `json:"path"`      // 路径
	Name      string         `json:"name"`      // 名称
	Component string         `json:"component"` // 组件
	Redirect  string         `json:"redirect"`  // 重定向
	Meta      map[string]any `json:"meta"`      // 元数据
	Icon      string         `json:"icon"`      // 图标
	Sort      int64          `json:"sort"`      // 排序
	Children  []*MenuItem    `json:"children"`
}

type MenuCreateReq struct {
	Path      string `json:"path" validate:"required"`
	Name      string `json:"name" validate:"required"`
	Component string `json:"component"`
	Redirect  string `json:"redirect"`
	Meta      string `json:"meta"`
	Icon      string `json:"icon"`
}

type MenuUpdateReq struct {
	ID        int64  `json:"id" validate:"required"`
	Path      string `json:"path"`
	Name      string `json:"name"`
	Component string `json:"component"`
	Redirect  string `json:"redirect"`
	Meta      string `json:"meta"`
	Icon      string `json:"icon"`
}

File: web/src/api/mgr/menu.ts
import { request } from '@/utils/request';
import type { PageResp } from '../model';

export interface MenuMeta {
  title?: Record<string, string>;
  [key: string]: any;
}

export interface MenuItem {
  id: number;
  parent_id?: number;
  menu_type?: number;
  path: string;
  name: string;
  component?: string;
  redirect?: string;
  meta?: MenuMeta;
  icon?: string;
  sort?: number;
  children?: MenuItem[];
  created_at?: number;
  updated_at?: number;
}

export interface MenuListReq {
  page?: number;
  page_size?: number;
  name?: string;
}

export interface MenuCreateReq {
  parent_id?: number;
  name: string;
  path: string;
  component?: string;
  icon?: string;
  sort?: number;
  meta?: string;
}

export interface MenuUpdateReq {
  id: number;
  parent_id?: number;
  name?: string;
  path?: string;
  component?: string;
  icon?: string;
  sort?: number;
  meta?: string;
}

const Api = {
  MenuList: '/menu/list',
  MenuTree: '/menu/tree',
  MenuCreate: '/menu',
  MenuUpdate: '/menu',
  MenuDelete: '/menu',
};

class MenuApi {
  list(params?: MenuListReq) {
    return request.get<PageResp<MenuItem>>({
      url: Api.MenuList,
      params,
    });
  }

  tree() {
    return request.get<MenuItem[]>({
      url: Api.MenuTree,
    });
  }

  create(data: MenuCreateReq) {
    return request.post<MenuItem>({
      url: Api.MenuCreate,
      data,
    });
  }

  update(data: MenuUpdateReq) {
    return request.put<MenuItem>({
      url: Api.MenuUpdate,
      data,
    });
  }

  delete(ids: number[]) {
    const params = { ids: ids.join(',') };
    return request.delete({
      url: Api.MenuDelete,
      params,
    });
  }
}

export const menuApi = new MenuApi();

